jhputnam/test_mailroom.py

```diff
@@ -0,0 +1,140 @@
+"""
+Mailroom testing.
+"""
+
+import pickle
+import pytest
+
+import mailroom
+
+
+def test_initdb_dbpresent():
+    """Test dbinit with db file present."""
+
+    testdb = {'Dan Rather': [3453.0], 'SpongeBob': [44.05]}
+
+    # Call initdb with test data.
+    donors = mailroom.initdb('testdb.dat')
+
+    # Assert that the test data is what we expect.
+    assert donors == testdb
+
+
+def test_initdb_dbmissing1(monkeypatch):
+    """Test dninit function with db missing and option 1."""
+
+    # Override input and supply option 1.
+    monkeypatch.setitem(__builtins__, 'input', lambda x: '1')
```

Comment:
Took me a while find any documentation on this.  Curious if you could share the docs you were using, but once i realized what this was doing my mind was blown.  I'm definitely going to remember this for the future :)